sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor{TPartition}.cs

```diff
@@ -0,0 +1,437 @@
+
+            // TODO: Implement
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="EventProcessor{TPartition}"/> class.
+        /// </summary>
+        ///
+        /// <param name="eventBatchSize">The desired number of events to include in a batch to be processed.  This size is the maximum count in a batch; the actual count may be smaller, depending on whether events are available in the Event Hub.</param>
+        /// <param name="consumerGroup">The name of the consumer group the processor is associated with.  Events are read in the context of this group.</param>
+        /// <param name="fullyQualifiedNamespace">The fully qualified Event Hubs namespace to connect to.  This is likely to be similar to <c>{yournamespace}.servicebus.windows.net</c>.</param>
+        /// <param name="eventHubName">The name of the specific Event Hub to associate the processor with.</param>
+        /// <param name="credential">The Azure managed identity credential to use for authorization.  Access controls may be specified by the Event Hubs namespace or the requested Event Hub, depending on Azure configuration.</param>
+        /// <param name="options">The set of options to use for the processor.</param>
+        ///
+        protected EventProcessor(int eventBatchSize,
+                                 string consumerGroup,
+                                 string fullyQualifiedNamespace,
+                                 string eventHubName,
+                                 TokenCredential credential,
+                                 EventProcessorOptions options = default)
+        {
+           // TODO: Implement
+           throw new NotImplementedException();
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="EventProcessor{TPartition}"/> class.
+        /// </summary>
+        ///
+        protected EventProcessor()
+        {
+        }
+
+        /// <summary>
+        ///   Signals the <see cref="EventProcessor{TPartition}" /> to begin processing events.  Should this method be called while the processor
+        ///   is running, no action is taken.
+        /// </summary>
+        ///
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> instance to signal the request to cancel the start operation.  This won't affect the <see cref="EventProcessor{TPartition}" /> once it starts running.</param>
+        ///
+        /// <exception cref="EventHubsException">Occurs when this <see cref="EventProcessor{TPartition}" /> instance is already closed; the <see cref="EventHubsException.FailureReason"/> will be set to <see cref="EventHubsException.FailureReason.ClientClosed"/>.</exception>
+        ///
+        public virtual async Task StartProcessingAsync(CancellationToken cancellationToken = default) =>
+            await StartProcessingInternalAsync(true, cancellationToken).ConfigureAwait(false);
```

Comment:
Yes, you'd have an asynchronous operation that was unobserved and which would be unable to surface exceptions; anything that blew up in there would go unobserved.

Since we've got naturally async operations going on, it doesn't make sense to force them to synchronous to avoid returning the task, which is what the sync versions do.